test/cpp_api_parity/functional_impl_check.py

```diff
@@ -0,0 +1,258 @@
+    if not hasattr(unit_test_class, 'functional_test_params_map'):
+        unit_test_class.functional_test_params_map = {}
+    for test_params_dict in test_params_dicts:
+        # Skip all `torch.nn` module tests, since they are handled by another test suite.
+        if not is_torch_nn_functional_test(test_params_dict):
+            continue
+
+        assert 'cpp_options_args' in test_params_dict or 'cpp_function_call' in test_params_dict, (
+            "To enable C++ API parity test, "
+            "`cpp_options_args` or `cpp_function_call` entry must be present in test params dict:\n{}. \n"
+            "If you are interested in adding the C++ API parity test, please see:\n"
+            "NOTE [How to check NN module / functional API parity between Python and C++ frontends]. \n"
+            "If not, please add `test_cpp_api_parity=False` to the test params dict and file an issue about this."
+        ).format(pprint.pformat(test_params_dict))
+
+        assert not ('cpp_options_args' in test_params_dict and 'cpp_function_call' in test_params_dict), (
+            "Only one of `cpp_options_args` and `cpp_function_call` entries "
+            "should be present in test params dict:\n{}").format(pprint.pformat(test_params_dict))
+
+        functional_name = compute_functional_name(test_params_dict)
+
+        assert hasattr(torch.nn.functional, functional_name), \
+            "`torch.nn.functional` doesn't have function `{}`. (Discovered while processing\n{}.)".format(
+                functional_name, pprint.pformat(test_params_dict))
+
+        functional_full_name = 'F::' + functional_name
+
+        assert functional_full_name in parity_table['torch::nn::functional'], (
+            "Please add `{}` entry to `torch::nn::functional` section of `test/cpp_api_parity/parity-tracker.md`. "
+            "(Discovered while processing\n{}.)").format(functional_full_name, pprint.pformat(test_params_dict))
+
+        for device in devices:
+            test_params = process_test_params_for_functional(
+                test_params_dict=test_params_dict,
+                device=device,
+                test_instance_class=test_instance_class,
+            )
+            test_name = 'test_torch_nn_functional_{}'.format(test_params.functional_variant_name)
+            unit_test_class.functional_test_params_map[test_name] = test_params
+
+            def test_fn(self):
+                test_torch_nn_functional_variant(
+                    unit_test_class=self, test_params=unit_test_class.functional_test_params_map[self._testMethodName])
+
+            test_fn = decorate_test_fn(
+                test_fn=test_fn,
+                test_cpp_api_parity=test_params_dict.get('test_cpp_api_parity', True),
+                test_cuda=test_params_dict.get('test_cuda', True),
+                has_impl_parity=parity_table['torch::nn::functional'][functional_full_name][0] and
```

Comment:
nit: this is pretty hard to read. It might be nicer if parity_table were a class with some nice methods on it to get things. for example,
```
parity_table.find('torch::nn::functional::' + functional_full_name).has_impl_parity
```
but that's not really necessary

Reply:
Agreed that it would make it easier to read. We can do it in a follow-up PR